packages/neo-one-client/src/user/LocalUserAccountProvider.ts

```diff
@@ -823,6 +841,172 @@ export class LocalUserAccountProvider<TKeyStore extends KeyStore, TProvider exte
+
+    const createConsolidatedTransaction = (): TransactionModel => {
+      switch (transactionUnsignedIn.type) {
+        case TransactionType.Contract:
+          return new ContractTransaction({
+            version: transactionUnsignedIn.version,
+            attributes: transactionUnsignedIn.attributes,
+            inputs: transactionUnsignedIn.inputs.concat(newInputs),
+            outputs: updatedOutputs.concat(newOutputs),
+            scripts: transactionUnsignedIn.scripts,
+            hash: transactionUnsignedIn.hash,
+          });
+        case TransactionType.Claim:
+          return new ClaimTransaction({
+            version: transactionUnsignedIn.version,
+            attributes: transactionUnsignedIn.attributes,
+            inputs: transactionUnsignedIn.inputs.concat(newInputs),
+            outputs: updatedOutputs.concat(newOutputs),
+            scripts: transactionUnsignedIn.scripts,
+            hash: transactionUnsignedIn.hash,
+            claims: transactionUnsignedIn.claims,
+          });
+        case TransactionType.Issue:
+          return new IssueTransaction({
+            version: transactionUnsignedIn.version,
+            attributes: transactionUnsignedIn.attributes,
+            inputs: transactionUnsignedIn.inputs.concat(newInputs),
+            outputs: updatedOutputs.concat(newOutputs),
+            scripts: transactionUnsignedIn.scripts,
+            hash: transactionUnsignedIn.hash,
+          });
+        default:
+          return transactionUnsignedIn;
+      }
+    };
+
+    const getMessageSize = () => messageSize + newInputs.length * newInputSize + newOutputs.length * newOutputSize;
+    const updateOutput = (output: OutputModel, amount: BigNumber): OutputModel =>
+      new OutputModel({
+        asset: output.asset,
+        value: output.value.add(clientUtils.bigNumberToBN(amount, 8)),
+        address: output.address,
+      });
+
+    const consoladatableUnspentsIn = await this.provider.getUnspentOutputs(from.network, from.address, monitor);
+    consoladatableUnspentsIn
+      .filter(
+        (unspent) =>
+          !inputs.some((input) => unspent.txid === input.txid && unspent.vout === input.vout) &&
+          !this.mutableUsedOutputs.has(`${unspent.txid}:${unspent.vout}`),
+      )
+      .forEach((unspent) => {
+        if ((looseAssets[unspent.asset] as ReadonlyArray<UnspentOutput> | undefined) !== undefined) {
+          looseAssets = {
+            ...looseAssets,
+            [unspent.asset]: looseAssets[unspent.asset].concat([unspent]),
+          };
+        } else {
+          looseAssets = {
+            ...looseAssets,
+            [unspent.asset]: [unspent],
+          };
+        }
+      });
+
+    const returnOutputs = transactionUnsignedIn.outputs.filter(
+      (output) => common.uInt160ToString(output.address) === addressToScriptHash(from.address),
+    );
+    let newUnspents: ReadonlyArray<UnspentOutput>;
+
+    if (returnOutputs.length > 0) {
```

Comment:
If `returnOutputs.length === 0` the following lines are a no-op, so you don't need to check that it's larger.